coeus-impl/src/main/java/org/kuali/coeus/common/budget/impl/period/BudgetPeriodRule.java

```diff
@@ -340,6 +340,10 @@ private boolean isValidNewBudgetPeriod(Budget budget, BudgetPeriod newBudgetPeri
                         newBudgetPeriod.setBudgetPeriod(periodNum+1);
                     }
                     break;
+                }else if(newPeriodStartDate.compareTo(periodEndDate) <= 0) {
+                    dateCompareValue = "ERROR_NEW_PERIOD_START_BEFORE_PREVIOUS_END";
+                    saveErrors(dateCompareValue, errorMap, newPeriodDateParams);
+                    validNewBudgetPeriod = false;
```

Comment:
So if you edit the budget periods to create a gap between two existing budget periods. Say 1/1/2014-6/30/2014 and 1/1/2015-6/30/2015. And then try to insert a new budget period at 7/1/2014-12/31/2014. Does that work? Seems as though this change would cause that to fail when it shouldn't. Am I reading the logic here wrong?